Server/ReasnAPI/ReasnAPI/Services/EventService.cs

```diff
@@ -1,16 +1,276 @@
-﻿using ReasnAPI.Models.Database;
-
-namespace ReasnAPI.Services {
-    public class EventService (ReasnContext context) {
-        private readonly ReasnContext _context = context;
-
-        /* TODO: Create following functions for this class
-         * create
-         * update
-         * delete
-         * get by ID
-         * get list by filter
-         * get all
-         */
+﻿using Microsoft.EntityFrameworkCore;
+using ReasnAPI.Models.Database;
+using ReasnAPI.Models.DTOs;
+using System.Linq.Expressions;
+using System.Linq;
+using System.Transactions;
+
+namespace ReasnAPI.Services;
+public class EventService(ReasnContext context)
+{
+    public EventDto CreateEvent(EventDto eventDto)
+    {
+        using (var scope = new TransactionScope())
+        {
+            eventDto.Slug = CreateSlug(eventDto);
+
+            var newEvent = new Event
+            {
+                Name = eventDto.Name,
+                AddressId = eventDto.AddressId,
+                Description = eventDto.Description,
+                OrganizerId = eventDto.OrganizerId,
+                StartAt = eventDto.StartAt,
+                EndAt = eventDto.EndAt,
+                CreatedAt = DateTime.Now,
+                UpdatedAt = DateTime.Now,
+                Slug = eventDto.Slug,
+                StatusId = eventDto.StatusId,
+            };
+
+            context.Events.Add(newEvent);
+            context.SaveChanges();
+
+            var eventId = newEvent.Id;
+            if (eventDto.Tags == null)
+            {
+                return eventDto;
+            }
+
+            foreach (var tag in eventDto.Tags)
+            {
+                var newTag = context.Tags.FirstOrDefault(r => r.Name == tag.Name);
+                if (newTag == null)
+                {
+                    var tagToAdd = new Tag { Name = tag.Name };
+                    context.Tags.Add(tagToAdd);
+                    context.SaveChanges();
+
+                    var eventTag = new EventTag
+                    {
+                        EventId = eventId,
+                        TagId = tagToAdd.Id
+                    };
+                    context.EventTags.Add(eventTag);
+                    context.SaveChanges();
+                }
+                else
+                {
+                    var eventTag = new EventTag
+                    {
+                        EventId = eventId,
+                        TagId = newTag.Id
+                    };
+                    context.EventTags.Add(eventTag);
+                    context.SaveChanges();
+                }

```

Comment:
Can adding new eventTag be done once? After if, we can then do something like newTag = new Tag { Name = tag.Name }; instead of new variable tagToAdd and then do adding new EventTag once, after if (since newTag will be either got from DbContext or the one created)